src/Server/EmbeddedClient/openpty.cpp

```diff
@@ -0,0 +1,83 @@
+#include <cstdlib>
+#include <cstring>
+#include <fcntl.h>
+#include <termios.h>
+#include <unistd.h>
+#include <Server/EmbeddedClient/openpty.h>
+#include <sys/ioctl.h>
+#ifdef __FreeBSD__    // Not supported by FreeBSD version in sysroot https://www.gnu.org/software/gnulib/manual/html_node/ptsname_005fr.html
+#include <cstdio>
+
+
+static int ptsname_r(int fd, char * buf, size_t buflen)
```

Comment:
Is this code copypasted from somewhere? Asking because it doesn't look very ClickHouse-y. If yes, we should provide the source.

Reply:
I remember writing it myself based on some libc samples